src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -2083,87 +2085,114 @@ void InstrumentClipView::adjustVelocity(int32_t velocityChange) {
+						// Sean: check for pop-up so that you don't change encoder turn (cause you may just want to see
+						// the value)
 						noteRow->changeNotesAcrossAllScreens(note->pos, modelStackWithNoteRow, action,
 						                                     CORRESPONDING_NOTES_ADJUST_VELOCITY, velocityChange);
 					}
 
-					if (velocityValue == 0) {
-						velocityValue = note->getVelocity();
-					}
-					else {
-						if (velocityValue != note->getVelocity()) {
-							velocityValue = 255; // Means "multiple"
-						}
-					}
+					getVelocityValue(velocityValue, note->getVelocity());
+
 					numNotesThisSquare++;
 					velocitySumThisSquare += note->getVelocity();
 
 					noteI++;
 					note = noteRow->notes.getElement(noteI);
 				}
 
-				editPadPresses[i].intendedVelocity =
-				    velocitySumThisSquare / numNotesThisSquare; // Get the average. Ideally we'd have done this when
-				                                                // first selecting the note too, but I didn't
+				// Rohan: Get the average. Ideally we'd have done this when first selecting the note too, but I didn't
+
+				// Sean: not sure how getting the average when first selecting the note would help because the average
+				// will change based on the velocity adjustment happening here.
+
+				// We're adjusting the intendedVelocity here because this is the velocity that is used to audition
+				// the pad press note so you can hear the velocity changes as you're holding the note down
+				editPadPresses[i].intendedVelocity = velocitySumThisSquare / numNotesThisSquare;
 			}
 
 			// Only one note in square
 			else {
 				if (display->hasPopup()) {
+					// Sean: We're adjusting the intendedVelocity here because this is the velocity that is used to
+					// audition the pad press note so you can hear the velocity changes as you're holding the note down
 					editPadPresses[i].intendedVelocity =
 					    std::clamp<int32_t>((int32_t)editPadPresses[i].intendedVelocity + velocityChange, 1, 127);
 					noteRow->changeNotesAcrossAllScreens(editPadPresses[i].intendedPos, modelStackWithNoteRow, action,
 					                                     CORRESPONDING_NOTES_ADJUST_VELOCITY, velocityChange);
 				}
 
-				if (velocityValue == 0) {
-					velocityValue = editPadPresses[i].intendedVelocity;
-				}
-				else {
-					if (velocityValue != editPadPresses[i].intendedVelocity) {
-						velocityValue = 255; // Means "multiple"
-					}
-				}
+				getVelocityValue(velocityValue, editPadPresses[i].intendedVelocity);
 			}
 		}
 	}
 
+	displayVelocity(velocityValue, velocityChange);
+
+	reassessAllAuditionStatus();
+}
+
+// determines whether or not you're trying to adjust the velocities of multiple notes
+// with different starting velocities (prior to adjustment)
+// used to determine whether to display the updated velocity value or a generalized
+// "velocity increased / decreased" message
+void InstrumentClipView::getVelocityValue(int32_t& velocityValue, int32_t velocity) {
```

Comment:
Nit:

For better clarity the name should be updateVelocityValue and the second argument should be newVelocity

Should have a doc comment too like
///sets velocityValue to newvolicity if velocityValue is 0. Keep it at velocityValue if they're the same. Set it to 255 if they're different